src/clients/bonus.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  ApiHeaderJson,
+  composeHeaderProducers,
+  createFetchRequestForApi,
+  ReplaceRequestParams,
+  RequestHeaderProducer,
+  RequestParams,
+  TypeofApiCall
+} from "italia-ts-commons/lib/requests";
+import { Omit } from "italia-ts-commons/lib/types";
+import nodeFetch from "node-fetch";
+
+import {
+  startBonusEligibilityCheckDefaultDecoder,
+  StartBonusEligibilityCheckT
+} from "../../generated/io-bonus-api/requestTypes";
+
+// we want to authenticate against the platform APIs with
+// the Azure Functions header key
+function SubscriptionKeyHeaderProducer<P>(
+  token: string
+): RequestHeaderProducer<P, "Ocp-Apim-Subscription-Key"> {
+  return () => ({
+    "Ocp-Apim-Subscription-Key": token,
+    "X-Functions-Key": token
+  });
+}
+
+export function APIClient(
```

Comment:
we can call it `BonusAPIClient`? To avoid naming conflict with the already existing `APIClient`